src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -387,6 +389,79 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructorReminders(
+            if (student != null) {
+                String additionalContactInformation = HTML_NO_ACTION_REQUIRED
+                        + getAdditionalContactInformationFragment(course);
+
+                String submitUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_SUBMISSION_EDIT_PAGE)
+                        .withCourseId(course.getId())
+                        .withSessionName(session.getFeedbackSessionName())
+                        .withStudentEmail(userEmail)
+                        .toAbsoluteString();
+
+                String reportUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_RESULTS_PAGE)
+                        .withCourseId(course.getId())
+                        .withSessionName(session.getFeedbackSessionName())
+                        .withStudentEmail(userEmail)
+                        .toAbsoluteString();
+
+                if (session.isClosed()) {
+                    buffer.append("<br>")
+                            .append(Templates.populateTemplate(template,
+                                    "${courseName}", SanitizationHelper.sanitizeForHtml(course.getName()),
+                                    "${courseId}", SanitizationHelper.sanitizeForHtml(course.getId()),
+                                    "${feedbackSessionName}",
+                                    SanitizationHelper.sanitizeForHtml(session.getFeedbackSessionName()),
+                                    "${deadline}", SanitizationHelper.sanitizeForHtml(session.getEndTimeString()),
+                                    "${instructorFragment}", "",
+                                    "${sessionInstructions}", session.getInstructionsString(),
+                                    "${feedbackAction}", feedbackAction,
+                                    "${additionalContactInformation}", additionalContactInformation));
+                } else {
+                    buffer.append("<br>")
+                            .append(Templates.populateTemplate(template,
+                                    "${courseName}", SanitizationHelper.sanitizeForHtml(course.getName()),
+                                    "${courseId}", SanitizationHelper.sanitizeForHtml(course.getId()),
+                                    "${feedbackSessionName}",
+                                    SanitizationHelper.sanitizeForHtml(session.getFeedbackSessionName()),
+                                    "${deadline}", SanitizationHelper.sanitizeForHtml(session.getEndTimeString()),
+                                    "${instructorFragment}", "",
+                                    "${sessionInstructions}", session.getInstructionsString(),
+                                    "${submitUrl}", submitUrl,
+                                    "${reportUrl}", reportUrl,
+                                    "${feedbackAction}", feedbackAction,
+                                    "${additionalContactInformation}", additionalContactInformation));
+                }
```

Comment:
The `if` `else` only differ by `submitUrl` and `reportUrl` right? Can minimize repeating the code here?

Reply:
Anyway a closed session may still have `reportUrl`?

Reply:
Then maybe it's better to have `submitUrl` and `reportUrl` on both templates, and give empty or some explanatory text for `submitUrl` for closed session?